multiplexer/appd/run_test.go

```diff
@@ -1,12 +1,47 @@
+//go:build multiplexer
+
 package appd
 
 import (
+	"bytes"
+	"github.com/celestiaorg/celestia-app/v4/internal/embedding"
 	"os"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 )
 
+// TestCreateExecCommand execs a command to an embedded binary.
+func TestCreateExecCommand(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test which expects an embedded binary")
+	}
+
+	bin, err := embedding.CelestiaAppV3()
+	require.NoError(t, err)
+
+	appdInstance, err := New("test-app", bin)
+	require.NoError(t, err)
+	require.NotNil(t, appdInstance)
+
+	cmd := appdInstance.CreateExecCommand("version")
+	require.NotNil(t, cmd)
+
+	var outputBuffer bytes.Buffer
+	cmd.Stdout = &outputBuffer
+
+	require.NoError(t, cmd.Run())
+
+	output := outputBuffer.String()
+	require.NotEmpty(t, output)
+	require.True(t, strings.HasPrefix(output, "3"), "major version should be 3 of embedded binary. Got: %s", output)
+
+	appdInstance.cleanup()
```

Comment:
this cleanup fn is never actually used anywhere, meaning right now there is a constant bloat of binaries. Ran this locally and saw that I have  84 in this directory.

```
ls /var/folders/sw/2yv0fv911xg13xpppjk3sc080000gn/T/  | grep appd | wc -l
84
```

Reply:
Yes 100% agreed. The bloat increases each time `celestia-appd` is invoked.